### ISSUE TITLE
Badge Request Cloudflare

### ISSUE THREAD

**Issue by planetahuevo** [badges/shields#2714]
:clipboard: **Description**

- Which service is this badge for: Cloudflare
- What sort of information should this badge show?
If the website is using cloudflare or not. 
Cloudflare | active
Cloudflare | offline

:link: **Data**

Where can we get the data from?

- Is there a public API?
Yes, there is.
- Does the API requires an API key?
Yes. 
- Link to the API documentation.
https://api.cloudflare.com/

I have limited technical knowledge so I am not sure if the API allows to create the badge I am looking for, but I can see the documentation is huge and I am sure some other badges could be created from this one.

Thanks!

**Comment by paulmelnikow** [comment]
Hi! Thanks for the suggestion. Can you explain a little more about why this badge should be implemented and how it would be used?

**Comment by calebcartwright** [comment]
It looks like the API requires account specific cred info to be sent with the request: which would make this one tricky outside of self-hosted Shields instances.

https://api.cloudflare.com/#getting-started-requests

>Requests
Requests must be sent over HTTPS with any payload formatted in JSON. All requests must include both  and  headers to authenticate. Requests that use  can use that instead of the Auth-Key and Auth-Email headers.

Name | Format | Description
-- | -- | --
API Key | X-Auth-Key | API key generated on the "My Account" page
Email | X-Auth-Email | Email address associated with your account
User Service Key | X-Auth-User-Service-Key | A special Cloudflare API key good for a restricted set of endpoints. Always begins with "v1.0-", may vary in length.

**Comment by planetahuevo** [comment]
Oh, not sure I understand that completely but I found also this one: 
https://www.cloudflarestatus.com/api
it is a different thing, as it will be just a badge for the status of their services on different points of the world, but could be a nice badge too. 
Which is really a statuspage badge, and maybe that is a complete different thing.

**Comment by Prouser123** [comment]
We can't really implement those 'using Cloudflare' badges, but we could have, for example, a CloudFlare category and implement some monitors from CloudflareStatus, like the CDN regions or various core CloudFlare features.

**Comment by jhabdas** [comment]
@planetahuevo you wouldn't happen to be leveraging the fact #1880 added cookies to hundreds of thousands of websites and leveraging that fact to find out who might not be in compliance with GDPR cookie consent would you? FYI: #2986

**Comment by jhabdas** [comment]
@planetahuevo you wouldn't happen to be leveraging the fact #1880 added cookies to hundreds of thousands of websites and using it to find out who might not be in compliance with GDPR cookie consent now would you? FYI: #2986

**Comment by planetahuevo** [comment]
I do not understand what do you mean at all. Can you rephrase it? 
Thanks

**Comment by jhabdas** [comment]
It's all good. Just trying to connect some dots that probably aren't there. 👍

**Comment by calebcartwright** [comment]
Hi @jhabdas. We'll be happy to discuss your concerns with CloudFlare over on #2986. 

However, please do not attack other members of the Shields community, or accuse them of malicious intent, especially without any supporting evidence. 

I'm going to mark these comments as off topic, and we can continue the dialog in the other thread
